### PR TITLE
BLOGS-1272 - Morph Comments

### PR DESCRIPTION
**Pull request by felipeparaujo**

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: bin/symlinks [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#!/bin/sh
```
Comment (votemike) [inline bin/symlinks]: Symlinking dependencies is grim 🤢

File: composer.json [hunk 2]
```diff
@@ -14,6 +14,10 @@
         {
             "type": "vcs",
             "url": "git@github.com:bbc/gel-iconography-assets.git"
+        },
+        {
+            "type": "vcs",
+            "url": "git@github.com:bbc/programmes-morph-library.git"
```
Comment (votemike) [inline composer.json]: Is this the final name we've settled on for this library? Do we want it prefixed with "programmes"?

File: config/services.yaml [hunk 3]
```diff
@@ -1,13 +1,15 @@
 # Put parameters here that don't need to change on each machine where the app is deployed
 # https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration
 parameters:
-    app.api_endpoint: 'https://api.live.bbc.co.uk/isite2-content-reader'
+    app.api_endpoint: 'https://api.test.bbc.co.uk/isite2-content-reader'
```
Comment (votemike) [inline config/services.yaml]: According to this PR: https://github.com/bbc/blogs-frontend/commit/2d9fe420629d3d021d972fe1a04ebe7833239926 we want to default to LIVE settings. So I think these should be left pointing to LIVE.

File: src/Controller/BlogsBaseController.php [hunk 4]
```diff
@@ -23,6 +24,7 @@ public static function getSubscribedServices()
     {
         return array_merge(parent::getSubscribedServices(), [
             TagService::class,
+            CommentsService::class,
```
Comment (votemike) [inline src/Controller/BlogsBaseController.php]: Do we want this in ALL the controllers? Wouldn't it be better to put it in the specific controller that need it?

File: src/Ds/Core/Module/_comments.scss [hunk 5]
```diff
@@ -0,0 +1,20 @@
+#comments a {
```
Comment (votemike) [inline src/Ds/Core/Module/_comments.scss]: Nest the a, ul, li and input all with a single #comments?

File: src/Service/CommentsService.php [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+    public function queuePostComments(Blog $blog, Post $post): void
+    {
+        $this->client->queueView(
+            'bbc-morph-comments-view',
+            [
+                'apiKey' => $this->apiKey,
+                'mode' => 'embedded',
+                'idctaEnv' => $this->env,
+                'forumId' => 'blogs_' . $blog->getId() . $post->getForumId(),
```
Comment (votemike) [inline src/Service/CommentsService.php]: A private method to create this?

File: bin/symlinks (hunk 1 above)
Comment (felipeparaujo) [reply bin/symlinks]: it also eases development. i don't want to push to github and run a composer update for every single change I make in programmes-morph-library. Also, we have this for all the other projects

File: composer.json (hunk 2 above)
Comment (felipeparaujo) [reply composer.json]: It doesn't matter, but I think it should. It's supposed to be used only by us, and it makes it easier to find when we search for our repos in github

File: config/services.yaml (hunk 3 above)
Comment (felipeparaujo) [reply config/services.yaml]: oh yeah, my bad. It forgot to change it back

**Pull request merged**